pygac/calibration.py

```diff
@@ -455,21 +174,56 @@ def calibrate_solar(counts, chan, year, jday, spacecraft, corr=1):
     return refl
 
 
-def calibrate_thermal(counts, prt, ict, space, line_numbers, channel, spacecraft):
-    """Do the thermal calibration and return brightness temperatures (K)."""
-    cal = Calibrator(spacecraft)
+def calibrate_thermal(counts, prt, ict, space, line_numbers, channel, spacecraft, custom_coeffs=None):
+    """Do the thermal calibration and return brightness temperatures (K).
+
+    Arguments:
+        counts (array) - counts for the given channel
+        prt (array) - counts of the Platinum Resistance Thermometers (PRT)
+        ict (array) - counts of the In-orbit Calibration Targets (ICT)
+        space (array) - counts of cold space
+        line_numbers (array) - line number index
+        channel (array) - pygac internal channel index array
+        spacecraft (str) - pygac internal spacecraft name
+
+    Optionals:
+        custom_coeffs (dict) - custom calibration coefficients (default = None)
 
+    Note:
+        This function follows steps 1 to 4 from the KLM guide section 7.1.2.4
+        "Steps to Calibrate the AVHRR Thermal Channels"
```

Comment:
_in addition, I would add here that the KLM approach includes the approach by Walton et al. (1998), which is a correction for the non-linearity of channel 4 and 5. And also for the complete overview the smoothing approach by Trishchenko (2002) to account for fluctuations (outliers) in PRT measurements (see comment line 208)._